packages/example/src/screens/HeadlessCheckoutWithRedirect.tsx

```diff
@@ -29,98 +29,146 @@ const HeadlessCheckoutWithRedirect = (props: any) => {
-      //@ts-ignore
-      onRetrieving: data => {
-        const log = `\nonRetrieved: ${JSON.stringify(data)}\n`;
-        console.log(log);
-        setIsLoading(true);
-      },
-      //@ts-ignore
-      onRetrieved: data => {
-        const log = `\nonRetrieved: ${JSON.stringify(data)}\n`;
-        console.log(log);
-        setBanks(data);
-        setIsLoading(false);
-      },
-      //@ts-ignore
-      onValid: data => {
-        const log = `\nonValid: ${JSON.stringify(data)}\n`;
-        console.log(log);
-        setIsLoading(false);
-        setIsValidating(null);
-        if (data?.id) {
-          onSubmit()
-        }
-        // handleGoBack();
-      },
-      //@ts-ignore
-      onInvalid: data => {
-        const log = `\nonInvalid: ${JSON.stringify(data)}\n`;
-        console.log(log);
-        setIsLoading(false);
-        setIsValidating(null);
-      },
-      //@ts-ignore
-      onError: data => {
-        const log = `\nonError: ${JSON.stringify(data)}\n`;
-        console.log(log);
-        setIsLoading(false);
-        setIsValidating(null);
-      },
-    });
-  };
-
-  const onSubmit = async () => {
+  const submit = async () => {
     try {
       await redirectManager.submit();
     } catch (err) {
       console.error(err);
     }
   };
 
-  const pay = async (id: string) => {
-    try {
-      setIsValidating(id);
-      await redirectManager.onBankSelected(id);
-    } catch (err) {
-      setIsValidating(null);
-      console.error(err);
-    }
-  };
+  return (
+    <View
+      style={{
+        paddingHorizontal: 5,
+        flex: 1,
+        backgroundColor: 'white',
+      }}>
+      <Search
+        search={search}
+        onSearch={async (value: string) => {
+          setSearch(value);
+
+          try {
+            await redirectManager.onBankFilterChange(value);
+          } catch (err) {
+            console.error(err);
+          }
+        }}
+      />
+      <Banks
+        banks={banks}
+        isValidating={isValidating}
+        onPay={async (id: string) => {
+          try {
+            setIsValidating(id);
+            await redirectManager.onBankSelected(id);
+          } catch (err) {
+            setIsValidating(null);
+            console.error(err);
+          }
+        }}
+      />
+      <Loader isLoading={isLoading} />
+    </View>
+  );
+};
+
+
+const Search = ({ search, onSearch }: any) => {
+  return (
+    <TextField
+      title="Choose your bank"
+      textInputStyle={{
+        backgroundColor: '#f5f5f5',
+        borderColor: '#f5f5f5',
+        paddingHorizontal: 10,
+      }}
+      style={{
+        padding: 5,
+        marginVertical: 10,
+        borderRadius: 5,
+        marginHorizontal: 5,
+      }}
+      placeholder="Search Banks"
+      value={search}
+      onChangeText={(value) => {
+        onSearch(value);
+      }}
+    />
+  );
+}
+
+
+const Loader = ({ isLoading }: any) => {
+  if (!isLoading) {
+    return null;
+  } else {
```

Comment:
💅🏻 [nitpick] since the condition returns, we can omit the `else` to make it effectively a guard clause, which is clearer to read in my opinion, e.g.:

```ts
const Loader = ({ isLoading }: any) => {
  if (!isLoading) return null;
  
  return (
    <View
      // ...
    </View>
  );
};
```